plugins/outputs/zabbix/utils.go

```diff
@@ -0,0 +1,126 @@
+package zabbix
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/datadope-io/go-zabbix/v2"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+)
+
+var (
+	errMetricNoMeasurementName = fmt.Errorf("empty measurement name")
+	errMetricNoTags            = fmt.Errorf("metric without tags")
+	errMetricJustHostTag       = fmt.Errorf("metric with only host tag")
+)
+
+// getHostname returns the hostname from the tags, or the system hostname if not found.
+func getHostname(hostTag string, tags []*telegraf.Tag) (string, error) {
+	for _, tag := range tags {
+		if tag.Key == hostTag {
+			return tag.Value, nil
+		}
+	}
```

Comment:
```suggestion
func getHostname(hostTag string, metric *telegraf.Metric) (string, error) {
	if hostname, ok := metric.GetTag(hostTag); ok {
		return hostname, nil
	}
```

Reply:
Fixed others parts of the code in db1264954